### PR TITLE
Remove use_quay_for_containers_stage

### PR DESCRIPTION
**Pull request by CryptoRodeo**
This is no longer an attribute of product versions, so we should remove the ability to configure this.

JIRA: CWFCONF-4867

### CODE REVIEW

**Comment by ktdreyer** [comment]
Thanks for the PR!

I'd like to keep accepting this parameter and unconditionally `pop()` it (make it a no-op.)

**Comment by CryptoRodeo** [comment]
Sounds good, do we also keep [these lines](https://github.com/ktdreyer/errata-tool-ansible/pull/286/files#diff-28d0871028cb16a3e645ee6520e2851eb0ac3f39f72502a0c6842b40b612ce88L55-L64) in a situation like this where the field is no longer accepted by ET?

**Comment by ktdreyer** [comment]
> Sounds good, do we also keep [these lines](https://github.com/ktdreyer/errata-tool-ansible/pull/286/files#diff-28d0871028cb16a3e645ee6520e2851eb0ac3f39f72502a0c6842b40b612ce88L55-L64) in a situation like this where the field is no longer accepted by ET?
Whoops, I had a comment there but forgot to submit it. I've submitted it now. The summary is that we should document that this field is deprecated and tell users to remove it.

File: library/errata_tool_product_version.py [hunk 1]
```diff
@@ -52,16 +52,6 @@
...
-   use_quay_for_containers_stage:
-     description:
-       - If enabled, the Errata Tool will instruct pub to push this product
-         version's container advisories to quay.io instead of docker-pulp for
-         stage pushes.
-       - Leave this unchecked unless this product version is ready
-         for Quay.io.
-     choices: [true, false]
-     required: false
-     default: false
```
[the remark this answers is not in the document]
Comment (CryptoRodeo) [reply library/errata_tool_product_version.py:64]: Done, thanks for the help with the description! I had to slightly edit it for flake8.

File: library/errata_tool_product_version.py [hunk 2]
```diff
@@ -340,8 +325,6 @@ def run_module():
     # value if the user has omitted it.
     if params['use_quay_for_containers'] is None:
         params.pop('use_quay_for_containers')
-    if params['use_quay_for_containers_stage'] is None:
```
[the remark this answers is not in the document]
Comment (CryptoRodeo) [reply library/errata_tool_product_version.py:343]: Done, added a comment for some more context.

File: tests/integration/errata_tool_product_version/create-2.yml [hunk 3]
```diff
@@ -37,7 +36,6 @@
       - not attributes.is_rhel_addon
       - not attributes.is_server_only
       - attributes.use_quay_for_containers
-      - attributes.use_quay_for_containers_stage
```
[the remark this answers is not in the document]
Comment (CryptoRodeo) [reply tests/integration/errata_tool_product_version/create-2.yml:40]: Yeah it's gone from the JSON response.

File: tests/integration/errata_tool_product_version/create-1.yml [hunk 4]
```diff
@@ -36,7 +36,6 @@
       - not attributes.use_quay_for_containers
       - not attributes.is_server_only
       - not attributes.use_quay_for_containers
-      - not attributes.use_quay_for_containers_stage
```
[the remark this answers is not in the document]
Comment (CryptoRodeo) [reply tests/integration/errata_tool_product_version/create-1.yml]: Ack, I re-added it then. Thanks for the context

**Comment by ktdreyer** [comment]
I've dropped py27 to get GitHub Actions to pass on this PR: https://github.com/ktdreyer/errata-tool-ansible/pull/287

**Comment by ktdreyer** [comment]
It's ok to skip unit testing this since it's just a removal, and we need to get this in now to unbreak users (including Ceph :)

I've updated the integration test GitHub runner to the latest errata-rails.git codebase (4.29-0-32-g0a478ed) so we can test against a modern ET instance. I've also copied your `CryptoRodeo:cwfconf-4687` branch to this repo so that GitHub will run the integration tests against it.

Here are the remaining changes to make:

- [ ] The [GitHub Action logs](https://github.com/ktdreyer/errata-tool-ansible/actions/runs/5601114290/jobs/10244512628?pr=286) show that we need to drop that `not attributes.use_quay_for_containers_stage` line from `tests/integration/errata_tool_product_version/create-1.yml`.
- [ ] You can remove the `if 'use_quay_for_containers_stage' in params:` handling from the `create_product_version()` method.

Other than that, this PR is looking good to me.

File: tests/integration/errata_tool_product_version/create-2.yml [hunk 5]
```diff
@@ -13,7 +13,6 @@
     allow_buildroot_push: true
     is_server_only: false
     use_quay_for_containers: true
-    use_quay_for_containers_stage: true
```
[the remark this answers is not in the document]
Comment (ktdreyer) [reply tests/integration/errata_tool_product_version/create-2.yml:16]: In fact, the purpose of this `create-2.yml` integration test is solely to exercise the `use_quay_for_containers` and `use_quay_for_containers_stage` parameters! Unfortunately I didn't document this at the top of the file, but you can discover it by comparing it to the first integration test, like:

```
diff -u tests/integration/errata_tool_product_version/create-1.yml tests/integration/errata_tool_product_version/create-2.yml
```

[the remark this answers is not in the document]
Comment (CryptoRodeo) [reply tests/integration/errata_tool_product_version/create-2.yml:16]: Re-added the line

**Comment by ktdreyer** [comment]
Thanks, I've pushed this branch to this repo so the integration tests can run on it. [They passed](https://github.com/ktdreyer/errata-tool-ansible/actions/runs/5602569897/jobs/10248003095?pr=286), so we can merge this.

**Pull request merged**